File: templates/problems/07_trees/01_invert_tree/python/solution.py
from helpers import TreeNode, Optional


def solve_invert_tree(root: Optional[TreeNode]) -> Optional[TreeNode]:
    """Invert a binary tree and return its root.

    Time: O(n), Space: O(h) where h is the height of the tree.
    """
    if root is None:
        return None
    root.left, root.right = root.right, root.left
    solve_invert_tree(root.left)
    solve_invert_tree(root.right)
    return root

File: templates/problems/07_trees/01_invert_tree/typescript/solution.ts
import { TreeNode } from "./helpers";

export function solveInvertTree(root: TreeNode | null): TreeNode | null {
    if (root === null) {
        return null;
    }
    const temp = root.left;
    root.left = root.right;
    root.right = temp;
    solveInvertTree(root.left);
    solveInvertTree(root.right);
    return root;
}
